src/test/java/nurseybook/logic/parser/ViewScheduleCommandParserTest.java

```diff
@@ -29,4 +31,21 @@ public void parse_validArgs_returnsViewScheduleCommand() {
                 new ViewScheduleCommand(new DateTimeContainsDatePredicate(keyDate), keyDate);
         assertParseSuccess(parser, "2021-11-02", expectedViewScheduleCommand);
     }
+
+    @Test
+    public void parse_invalidArgs_throwsParseException() {
+        assertParseFailure(parser, "a", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                ViewScheduleCommand.MESSAGE_USAGE));
+    }
+
+
+    @Test
+    public void parse_outOfBoundsKeyDate_throwsParseException() {
```

Comment:
Perhaps a test case can be written for a date that has passed?

Reply:
Good idea!